agr_java_core/src/main/java/org/alliancegenome/cache/CacheService.java

```diff
@@ -0,0 +1,138 @@
+package org.alliancegenome.cache;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.alliancegenome.api.entity.CacheStatus;
+import org.alliancegenome.core.config.CacheConfig;
+import org.alliancegenome.neo4j.view.View;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.type.CollectionType;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@RequestScoped
+public class CacheService {
+    
+    @Inject
+    private RemoteCacheManager manager;
+
+    public static ObjectMapper mapper = new ObjectMapper();
+
+    // Only the cacher should call this constructor
+    // In the context of an application server the manager will be injected
+    public CacheService() {
+        this.manager = CacheConfig.defaultRemoteCacheManager();
+    }
+
+    static {
+        mapper.disable(MapperFeature.DEFAULT_VIEW_INCLUSION);
+        //mapper.configure(MapperFeature.DEFAULT_VIEW_INCLUSION, false);
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        mapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+        mapper.enable(SerializationFeature.INDENT_OUTPUT);
+    }
+
+    public List<String> getAllKeys(CacheAlliance cacheSpace) {
+        RemoteCache<String, String> cache = getCacheSpace(cacheSpace);
+        //System.out.println(cache);
+        return new ArrayList<>(cache.keySet());
+    }
+
+    private RemoteCache<String, String> getCacheSpace(CacheAlliance cache) {
+        //log.info("Getting Cache Space: " + cache.getCacheName());
+        RemoteCache<String, String> remoteCache = manager.getCache(cache.getCacheName());
+        
+        if(remoteCache == null) {
+            return CacheConfig.createCache(cache);
+        }
+        
+        return remoteCache;
+    }
+
+    public <O> List<O> getCacheEntries(String entityID, CacheAlliance cacheSpace, Class<O> clazz) {
```

Comment:
What does the '0' in the generic syntax mean?

Reply:
It's used to specify the object that will be used in the params.... this way you don't have to use `<?>`

Reply:
Did not know that syntax. Thanks.